### ISSUE TITLE
Tests for twisted.test should conform to the new style of naming test classes

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7872]
|[<img alt="eeshangarg's avatar" src="https://avatars.githubusercontent.com/u/7251823?s=50" width="50" height="50">](https://github.com/eeshangarg)| @eeshangarg reported|
|-|-|
|Trac ID|trac#7872|
|Type|enhancement|
|Created|2015-04-26 13:28:44Z|
|Branch|https://github.com/twisted/twisted/tree/twisted-test-names-7872|

This ticket is a follow-up to ticket [ticket:7032].

Attachments:

* [misc_7872.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/e63/e63173f9526ad5706917ff722550b8df81b14784/91c2c0c6b1d653192d0c7cfe2fd86db4ec88c637.patch) (103004 bytes) - added by eeshangarg on 2015-04-26 14:28:29Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__7872 7872
type__enhancement enhancement
reporter__eeshangarg eeshangarg
priority__normal normal
milestone__ 
branch__branches_twisted_test_names_7872 branches/twisted-test-names-7872
branch_author__adiroiban adiroiban
status__closed closed
resolution__fixed fixed
component__core core
keywords__easy easy
time__1430054924847404 1430054924847404
changetime__1430064815067999 1430064815067999
version__None None
owner__eeshangarg eeshangarg
cc__eeshangarg
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="eeshangarg's avatar" src="https://avatars.githubusercontent.com/u/7251823?s=50" width="50" height="50">](https://github.com/eeshangarg)<a name="note_1"></a>|@eeshangarg removed owner|
|-|-|

Hello!

I am adding a patch for this (misc_7872.patch). This is one big diff because there were a lot of test classes in twisted.test that needed to be renamed. 

There were also cases where there were test classes like MessageParsingTestCase (which I renamed to MessageParsingTests) and MessageParsing2TestCase (for test classes named like this, I tried to give them more descriptive names based on what was being tested). 

Also there were test classes in the same file that went like TestFTPClient and FTPClientTestCase, now if you rename these two according to the FooTests naming convention recommended in our coding standard, you will end up with two classes named FTPClientTests in the same file, resulting in either one of them getting skipped and the tests for the skipped test class not being run at all. Here, again I tried to rename such classes and give them more descriptive names based on what or how a component was being tested. Here are the tests that I had to rename and how I renamed them:

- twisted.test.test_application.TestInternet2 <https://github.com/twisted/twisted/blob/trunk/twisted/test/test_application.py#L319> to simply InternetTests.

- twisted.test.test_defer.DeferredTestCaseII <https://github.com/twisted/twisted/blob/trunk/twisted/test/test_defer.py#L2059> to DeferredListEmptyTests.

- twisted.test.test_ftp.FTPClientTests <https://github.com/twisted/twisted/blob/trunk/twisted/test/test_ftp.py#L1422> to FTPClientFailedRETRAndErrbacksUponDisconnectTests. Also, see twisted.test.test_ftp.FTPClientTestCase <https://github.com/twisted/twisted/blob/trunk/twisted/test/test_ftp.py#L1475> which has been renamed as FTPClientTests.

- twisted.test.test_internet.CallFromThreadTests <https://github.com/twisted/twisted/blob/trunk/twisted/test/test_internet.py#L831> to CallFromThreadStopsAndWakeUpTests, so that CallFromThreadTestCase in the same file <https://github.com/twisted/twisted/blob/trunk/twisted/test/test_internet.py#L1042> can be renamed CallFromThreadTests.

- twisted.test.test_policies.TimeoutTestCase <https://github.com/twisted/twisted/blob/trunk/twisted/test/test_policies.py#L501> to TimeoutFactoryTests and TestTimeout in the same file <https://github.com/twisted/twisted/blob/trunk/twisted/test/test_policies.py#L642> to TimeoutMixinTests.

- twisted.test.test_sip.MessageParsingTestCase2 <https://github.com/twisted/twisted/blob/trunk/twisted/test/test_sip.py#L221> to MessageParsingFeedDataCharByCharTests. Also, see MessageParsingTestCase in the same file <https://github.com/twisted/twisted/blob/trunk/twisted/test/test_sip.py#L109> which has been renamed MessageParsingTests.

- twisted.test.test_tcp.HalfClose2TestCase <https://github.com/twisted/twisted/blob/trunk/twisted/test/test_tcp.py#L1557> to HalfCloseNoNotificationAndShutdownExceptionTests. Also, see HalfCloseTestCase in the same file <https://github.com/twisted/twisted/blob/trunk/twisted/test/test_tcp.py#L1475> which is now named HalfCloseTests.

And that's pretty much it, I guess. Let me know what you think! :-)

Regards,
Eeshan Garg

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban commented|
|-|-|

(In [44571]) Branching to twisted-test-names-7872.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set owner to @eeshangarg|
|-|-|

Hi,

Many thanks for this patch!

Changes looks good.

There is one minor problem which cause a py3 test to fail

```
twisted/test/test_internet.py:1038: undefined name 'Resolve'
```

I have fixed it and sent the changes for retesting.

Will merge if tests pass.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban set status to `closed`|
|-|-|

(In [44574]) Merge twisted-test-names-7872: Update test in twisted.test to conform to the new style of naming test classes.

Author: eeshangarg
Reviewers: adiroiban
Fixes: [#7872](https://github.com/twisted/twisted/issues/7872)
